db/db_impl/db_impl.h

```diff
@@ -1374,21 +1411,29 @@ class DBImpl : public DB {
   // not referenced in the MANIFEST (e.g.
   // 1. It's best effort recovery;
   // 2. The VersionEdits referencing the SST files are appended to
-  // MANIFEST, DB crashes when syncing the MANIFEST, the VersionEdits are
+  // RecoveryContext, DB crashes when syncing the MANIFEST, the VersionEdits are
   // still not synced to MANIFEST during recovery.)
-  // We delete these SST files. In the
+  // It stores the SST files to be deleted in RecoveryContext. In the
   // meantime, we find out the largest file number present in the paths, and
   // bump up the version set's next_file_number_ to be 1 + largest_file_number.
+  // recovery_ctx stores the context about version edits and files to be
+  // deleted. All those edits are persisted to new Manifest after successfully
+  // syncing the new WAL.
+  Status DeleteUnreferencedSstFiles(RecoveryContext* recovery_ctx);
   Status DeleteUnreferencedSstFiles();
 
-  // SetDbSessionId() should be called in the constuctor DBImpl()
-  // to ensure that db_session_id_ gets updated every time the DB is opened
-  void SetDbSessionId();
+  // Set DB identity file, and write DB ID to manifest if necessary.
+  Status SetDBId(bool read_only, RecoveryContext* recovery_ctx);
```

Comment:
I feel the diff-view of this PR can be made clearer if we keep `SetDBId()` and `SetDbSessionId()` the same as before. It will make it easier for other people to see what we actually do with these two functions.